core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -327,15 +365,18 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
     const auto reducer_wrapper = Experimental::Impl::make_sycl_function_wrapper(
         m_reducer, indirectReducerMem);
 
-    sycl_direct_launch(m_policy, functor_wrapper.get_functor(),
-                       reducer_wrapper.get_functor());
+    sycl::event event = sycl_direct_launch(
+        m_policy, functor_wrapper.get_functor(), reducer_wrapper.get_functor());
+    functor_wrapper.register_event(indirectKernelMem, event);
+    reducer_wrapper.register_event(indirectReducerMem, event);
   }
 
  private:
   FunctorType m_functor;
   Policy m_policy;
   ReducerType m_reducer;
   pointer_type m_result_ptr;
+  const bool m_result_ptr_device_accessible;
```

Comment:
Why const?
The `sycl_direct_launch` is const-qualified so it's not like it is going to be flipped by accident.  This qualifier implicitly deletes the copy-assignment operator which makes it harder to reason about that type.

Reply:
This also belongs to #4089, see https://github.com/kokkos/kokkos/pull/4089#discussion_r653181752.